### PR TITLE
Vertex AI Model Handler Private Endpoint Support

### PR DESCRIPTION
**Pull request by jrmccluskey**
Tightens retry logic to service-side errors and explicit 429s, explicitly handles private Vertex AI endpoints to support request sizes above 1.5 MB, adds some sanity-checking for args to the model handler (largely private + network pairing,) and makes the example pipeline compatible with private endpoints. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
Run Python_PVR_Flink PreCommit

**Comment by jrmccluskey** [comment]
R: @damccorm

File: sdks/python/apache_beam/examples/inference/vertex_ai_image_classification.py [hunk 1]
```diff
@@ -74,11 +74,24 @@ def parse_known_args(argv):
...
+      dest='vpc_network',
+      type=str,
+      required=False,
+      help='GCP network the endpoint is peered to')
   parser.add_argument(
       '--experiment',
       dest='experiment',
+      type=str,
       required=False,
       help='GCP experiment to pass to init')
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/inference/vertex_ai_image_classification.py:88]: Nit (missed previously): could we call out what experiment does here

**Review by damccorm — APPROVED** [review]
LGTM!

File: sdks/python/apache_beam/examples/inference/vertex_ai_image_classification.py (hunk 1 above)
Comment (jrmccluskey) [reply sdks/python/apache_beam/examples/inference/vertex_ai_image_classification.py:88]: I believe the experiment is passing a string to enable this feature - https://cloud.google.com/vertex-ai/docs/experiments/intro-vertex-ai-experiments

I can add the context link here, or I can drop it into the model handler itself. Or both, depending on preference. It's not wholly necessary for the example though

Comment (damccorm) [reply sdks/python/apache_beam/examples/inference/vertex_ai_image_classification.py:88]: I'd say both is most helpful. If we're offering the option, we should describe it

Comment (jrmccluskey) [reply sdks/python/apache_beam/examples/inference/vertex_ai_image_classification.py]: Wound up doing a little bit of both, better help message + a context link in the model handler docstring

**Pull request merged**